src/app/core/services/coursesService/courses.service.ts

```diff
@@ -1,125 +1,82 @@
 import { LimitByDatePipe } from "../../pipes/limit-by-date.pipe";
-import { Subscription, Observable, BehaviorSubject } from 'rxjs';
-
+import { Subscription, Observable, BehaviorSubject, Subject } from 'rxjs';
 
 @Injectable()
 export class CoursesService {
-	private courseListData: Course[];
-	public courseList: BehaviorSubject<Course[]>;
-	public courseListView: BehaviorSubject<Course[]>;
-	private courseListLimited:Course[];
-
-	constructor(private limitByDatePipe: LimitByDatePipe, private http: Http) {
+	private courseListData: any;
+	public courseList: Observable<Course[]>;
+	//public courseList: Subject<Course[]>;
+	public courseListV: BehaviorSubject<Course[]>;
+	private courseListLimited: Course[];
+	private courseListUrl: string = 'http://127.0.0.1:3004/courses?start=1&count=10';
+
+	constructor(private myLimitByDate: LimitByDatePipe, private http: Http) {
 		this.courseList = new BehaviorSubject([]);
-		this.courseListView = new BehaviorSubject([]);
-		this.courseListData = [
-			{
-				id: 0,
-				title: 'How to learn Angular 2 in few hours',
-				description: 'We all wanna get something without any effort, so forget about few hours, it will take you for ages.',
-				creationDate: new Date(2017, 3, 15),
-				duration: 167,
-				topRated: true
-			},
-			{
-				id: 1,
-				title: 'Yet another motivational video',
-				description: `See videos, read tons of articles, how to do at least anything, but advice is the one: just get your ass off the sofa and start doing.`,
-				creationDate: new Date(2017, 3, 7),
-				duration: 37,
-				topRated: false
-			},
-			{
-				id: 2,
-				title: 'The truth',
-				description: 'The truth is that we set up too big goals. They scares us. Fear has big eyes. Try to split them into small ones. 2 June',
-				creationDate: new Date(2017, 5, 2),
-				duration: 126,
-				topRated: true
-			},
-			{
-				id: 3,
-				title: 'The truth',
-				description: 'The truth is that we set up too big goals. They scares us. Fear has big eyes. Try to split them into small ones. 3 April',
-				creationDate: new Date(2017, 3, 3),
-				duration: 126,
-				topRated: true
-			}
-			,
-			{
-				id: 4,
-				title: 'The truth',
-				description: 'The truth is that we set up too big goals. They scares us. Fear has big eyes. Try to split them into small ones. 3 April',
-				creationDate: new Date(2017, 3, 4),
-				duration: 126,
-				topRated: true
-			}
-		];
+		//this.courseList = new Subject([]);
+		this.courseListV = new BehaviorSubject([]);
 	}
 
+	public getCourseItems():any {
 
-	public getCourseItems(): void {
-		this.courseListData = this.courseListData.map(item => {
-			console.log(item);
-			return {
-				id: item['id'],
-				title : item['title'],
-				description: item['description'],
-				date: item['creationDate'],
-				duration: item['duration'],
-				topRated: item['topRated']
-			}
-		});
+		return this.courseList
+			.switchMap((courseList) => this.http.get(this.courseListUrl))
```

Comment:
Maybe, you wanted to make here different logic:
1) Make request for courses
2) Transform result from server
3) put info to `this.courseList`
To do this, you would need to change current approach a little bit.
```javascript
// get courses
return this.http.get(this.courseListUrl)
   // make transformations
   .map( /* some logic which returns transformed data */ )
   // switch to behavior or replay subject to be able to subscribe in many places and get course list. 
   .switchMap( courseList => {
     /* set course list to subject and return subject */
   } )
```
Correct me, if I wrong.